packages/ui/src/utils/number-helpers.ts

```diff
@@ -127,3 +127,22 @@ export const formatAbbreviatedNumber = (
     suffix: "",
   };
 };
+
+/**
+ * Converts a number or string to a BigInt.
+ * @param value - The number or string to convert
+ * @param decimals - The number of decimal places (default is 18 for most ERC20 tokens)
+ * @returns A BigInt value representing the input with the appropriate decimal scaling
+ */
+export function numberToBigInt(value: number, decimals: number): bigint {
```

Comment:
Are we not doing a conversion like this anywhere else in the codebase already? Are we sure this is needed?